spec/openapi.yaml

```diff
@@ -68,6 +81,124 @@ components:
         - text
         - firmware
 
+    FileInfo:
+      type: object
+      properties:
+        type:
+          type: string
+          example: "PRINT_FILE"
+        refs:
+          type: object
+          properties:
+            download:
+              type: string
+              example: "/api/files/local/Spice_Harvester_0.3mm_PLA_MK3S_42m.gcode/raw"
+            icon:
+              type: string
+              example: "/api/thumbnails/local/Spice_Harvester_0.3mm_PLA_MK3S_42m.gcode.small.png"
+            thumbnail:
+              type: string
+              example: "/api/thumbnails/local/Spice_Harvester_0.3mm_PLA_MK3S_42m.gcode.orig.png"
+        size:
+          type: integer
+          example: 424242
+        path:
+          type: string
+          example: "/local/Spice_Harvester_0.3mm_PLA_MK3S_42m.gcode"
+        name:
+          type: string
+          example: "Spice_Harvester_0.3mm_PLA_MK3S_12m.gcode"
+        creation_timestamp:
+          type: integer
+          example: 1648038843
+        ro:
+          type: boolean
+          example: False
+        metadata:
+          type: object
+          properties:
+            bed_temperature:
+              type: integer
+              example: 60
+            nozzle_temperature:
+              type: integer
+              example: 215
+            brim_width:
+              type: number
+              example: 0
+            estimated printing time (normal mode):
+              type: string
+              example: "42m 42s"
```

Comment:
Whoa… a string? Should we have a prescribed format for it? Doesn't time format depend on locale? Isn't it better to have it in seconds and let the frontend decide on rendering?

And, do we have a field name with spaces and parentheses in it? :-O (this goes to other stuff below too).

Reply:
As i wrote above, it is value from g-code. So when the printer can process this attribute to something more computer-read great, but it can be next step.